ls1intum/Artemis
# [Feature] assess unsubmitted exams: programming exercises

### Checklist


### Motivation and Context



When using the `assess unsubmitted student exams` button in the student exams page, programming exercises where previously excluded from this.
![image](https://user-images.githubusercontent.com/33342534/124384808-4d6aff00-dcd3-11eb-8ee5-9948d1fc90ad.png) 
This PR changes that.

### Description

When an exam is not submitted, its programming submissions are now automatically assessed with 0 points too. 
Also when an exam was submitted, but no submission for a programming exercise, a submission is generated and again assessed with 0 points.

### Steps for Testing


1. Log in to Artemis
2. Create an exam -> create exercise groups, at least one of them containing a programming exercise
3. Add students
4. Student exams view: Create student exams and prepare exercises 
5. Participate with 4 students:
5.1 With one student use the submit button in the programming exercise, and hand in the exam.
5.2 With one student don't submit anything in the programming exercise, and hand in the exam.
5.3 With one student use the submit button in the programming exercise, and do not hand in the exam.
5.4. With one student don't use the submit button, and don't hand in the exam.
6. After the exam time is over, as an instructor go to the student exam view.
7. Press the 'assess unsubmitted student exams button'
8. Now from the exercise groups -> submissions page check if the assessment of all student exams was correct: All students, except the first one who submitted in the programming exercise, and submitted the exam, should now have assessed submissions. (With 0 points)

### Test Coverage



StudentExamService 87%
